### ISSUE TITLE
[BUG] 'symbols_loaded' event errors are too noisy

### ISSUE THREAD

**Issue by AlonZivony** [aquasecurity/tracee#2036]
## Prerequisites

- [ ] This affects latest released version.
- [x] This affects current development tree (origin/HEAD).
- [x] There isn't an issue describing the bug.

Select one OR another:

- [x] I'm going to create a PR to solve this (assign to yourself).
- [ ] Someone else should solve this.

## Bug description

The `symbols_loaded` event have errors resulting from the loader of symbols.
These errors happens frequently in some environments and from certain loads, and passed many times for each time there is error.

Example output:
failed to derive event 758: has no access to container fs - no recorded living process of mount namespace 4026532273
failed to derive event 758: has no access to container fs - no recorded living process of mount namespace 4026532273
failed to derive event 758: has no access to container fs - no recorded living process of mount namespace 4026532273
failed to derive event 758: has no access to container fs - no recorded living process of mount namespace 4026532273
failed to derive event 758: has no access to container fs - no recorded living process of mount namespace 4026532273
failed to derive event 758: has no access to container fs - no recorded living process of mount namespace 4026532273
failed to derive event 758: has no access to container fs - no recorded living process of mount namespace 4026532273
failed to derive event 758: has no access to container fs - no recorded living process of mount namespace 4026532273

## Steps to reproduce
Run the `symbols_loaded` event in environment with container already running.

**Comment by NDStrahilevitz** [comment]
This event is supposed to work with existing containers too right? Let's try to solve the root cause before just debug blocking the error, this could be a legit bug.

**Comment by AlonZivony** [comment]
Currently it doesn't suppose to work with existing containers.
The reason for that is that we use the `bucketcache` of tracee named `pidsInMntns` to get to the container path.
This cache is using the `sched_process_exec` event, so it is fed only with the creation of new containers.
@yanivagman wants to change this cache, so when it will be changed it should be better.

Anyways, there are many good reasons for this event to fail.
Because it is originated from user-space, the shared object or even the entire container might not be accessible anymore when the event is triggered.
Because I don't think there is a way to make sure we get rid of these errors, I think we should silence them (or at least give them lower logging level when we have a logger).

**Comment by NDStrahilevitz** [comment]
I understood the cause is the bucket cache, which, as far as i've checked, is populated when starting tracee. So it should work with old containers too, unless the populate skips them. So shouldn't the fix be pre-populating the cache correctly?

> Because it is originated from user-space, the shared object or even the entire container might not be accessible anymore when the event is triggered.

I see it as a good technical cause of failure, but as far as the result we would wish for, shouldn't we attempt to report the event anyway? And if not, then the event shouldn't fail to emit on error, but by design should return no derived event.



> Because I don't think there is a way to make sure we get rid of these errors, I think we should silence them (or at least give them lower logging level when we have a logger).

Maybe others not, but we don't know if other errors will happen yet. These errors seem like a bug in pre populating the cache wrong.

**Comment by AlonZivony** [comment]
> I understood the cause is the bucket cache, which, as far as i've checked, is populated when starting tracee. 

As far as I can see we are not doing any pre-population of the map currently.
We only inhibit the host mount namespace in the cache using the `/proc/1/ns/mnt` file.
We can enumerate over all the procfs, and it will help, but it will be very hard to find the first process of each container this way (maybe there are ways which I can't think of right now, I am not sure).

> I see it as a good technical cause of failure, but as far as the result we would wish for, shouldn't we attempt to report the event anyway? And if not, then the event shouldn't fail to emit on error, but by design should return no derived event.

I think we just don't have the way to handle less severe errors in tracee right now. This error should be logged somewhere, but not as critical one. Just as a debugging info (so if someone doesn't get the results he expect, he will be able to investigate why).

**Comment by rafaeldtinoco** [comment]
Alright, sorry for the late join to this discussion party. I wanted to read all the symbols_loaded event logic since its conception to better understand needs and this issue.

**TL;DR:**

There are inherent problems which we are already aware of since the code conception: The process which we're analyzing an event for might have already died (basically).

**Assumptions:**

- Technique here is to try to read filesystem from the procfs of a process in the same mnt namespace (we're iterating over existing pids, coming from the bucket cache). 

- The container (or last process in the mnt namespace) of the process of the event we're analyzing an event for might have already died. There is nothing we can do (thus this error).

- The buckets cache is not populated from existing mntns's, which means that any existing container won't be able to be analyzed for `symbols_loaded` derived event IF the process has died already.

**Some history:**

Considering we're buffering events in the perf-buffer, and then likely in the pipeline channels, and then, likely, by the caching code, it will be very easy to trigger the mentioned error for short-lived-containers (cache might take up to 2 minutes to be totally flushed, if you all recall performance tests for the caching mechanism).

Assuming we have a sustained throughput close to our limit (enough to fill the cache, but to allow it to be drained), then depending on the cache sizes (all of them) any `symbols_loaded` new event we will always be delayed by X seconds (lets assume from 30 sec to 1 min for the exercise ? Being conservative...).

With that said, it means that short-lived containers (bellow 1 min for this exercise) will likely fail to have the derived event successfully generated under higher loads (up to maximum theoretical sustained throughput).

**My opinion:**

1) NOW

The initial mount namespaces should be populated in buckets cache asap. It is very likely that we will have to use the "checking other mntns pids procfs directory" approach (kudos to Alon for the idea) when the pipeline is full. This is the line where we differentiate the introspection tool from the detection tool (a realistic environment for the detection tool might have a big amount of events, and delays in the pipeline, and already existing containers/pods running for hours, or days).

2) NOW

We are already loosing events because of other many reasons:

- perf-buffer might be overwritten due to high amount of sys events
- user-land might be capped by cpu limits causing delays
- OS overloaded could schedule tracee less frequently causing pipeline delays
- Virtual machine could have its vCPUs not scheduled often if running on VPSs

etc etc

In all those cases we keep a counter for errors and move on. I don't think this case should be different: do not generate the derived event, add a counter and move on.

3) FUTURE

WHEN we have the log printer in place (discussed before, suggested by Alon) then we can have a specific debug level for this (or a more detailed log level), for sure, but it will definitely be for "debugging purposes". I would even have some kind of "code area debug level" (that you can enable debug msgs from bucketscache, from pathresolver, from XXX package, etc).

**Comment by AlonZivony** [comment]
I need to open it because the existence of the errors is still problematic, because they are not really error in its deep meaning (because they are expected).
I will suppress all errors unless in debug, and when we have logger we will fix it.
